lib/crawler.js

```diff
@@ -1062,98 +1062,97 @@ Crawler.prototype.queueURL = function(url, referrer) {
 
 */
 Crawler.prototype.fetchQueueItem = function(queueItem) {
-    var crawler = this;
-
-    // Variable declarations
-    var clientRequest,
-        timeCommenced;
-
-    // Mark as spooled
-    queueItem.status = "spooled";
-    var client = queueItem.protocol === "https" ? https : http;
-    var agent  = queueItem.protocol === "https" ? crawler.httpsAgent : crawler.httpAgent;
+    var crawler = this,
+        timeCommenced = Date.now();
 
-    // Up the socket limit if required.
-    if (agent.maxSockets < crawler.maxConcurrency) {
-        agent.maxSockets = crawler.maxConcurrency;
-    }
+    crawler.queue.update(queueItem.url, {
+        status: "spooled"
+    }, function(error, queueItem) {
```

Comment:
I guess if we emit an error, and leave the queue item as it is, at some point the crawler will attempt to pick it up again (and if it's in the process of downloading — that is, the status as held by the queue is something other than `spooled` — it's the client's responsibility to keep an eye on the queue event to make sure that it's being handled appropriately. There's only so much we can do when the error handling strategy for the queue could be, well, anything.)